model/pipeline/src/main/resources/org/apache/beam/model/pipeline/pipeline.json

```diff
@@ -0,0 +1,320 @@
+
+        // After the watermark gets this far past the end of the window, new elements are dropped.
+        // in milliseconds
+        "allowedLateness": { "type": "number" }
+      }
+    },
+
+    "SideInput": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        // Only "multimap" is supported for initial Beam model
+        "accessPattern": { "type": "string" },
+
+        // The SDK converts the access pattern to an SDK-specific type, and this function
+        // converts it to the users desired side input type
+        "viewFn": { "type": "string" },
+
+        // A function that maps from the main input window to the desired side input window
+        "windowMappingFn": { "type": "string" }
+      }
+    },
+
+     // The parameters to a ParDo transform and further metatdata
+    "ParDoPayload": {
+      "type": "object",
+      "additionalProperties": false,
+
+      "properties": {
+        // The DoFn for this ParDo
+        "doFn": { "type": "string" },
+
+        // Whether the ParDo requires access to the current window (this may force runners to perform
+        // additional work prior to executing this ParDo)
+        "requiresWindowAccess": { "type": "boolean" },
+
+        // A map from PCollection id to the specification for how to materialize and read it
+        // as a side input
+        "sideInputs": {
+          "type": "object",
+          "additionalProperties": { "$ref" : "#/definitions/SideInput" }
+        }
+      }
+    },
+
+    // The parameters to a Read transform and further metadata.
+    //
+    // TODO: decide what metadata should be here in runner-readable form vs what is part of the
+    // UDF.
+    "ReadPayload": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+
+        // The ID of the user-definable source fn
+        "source": { "type": "string" },
+
+        "isBounded": { "type": "boolean" },
+        "doesNotRequireSplitting": { "type": "boolean" },
+        "producesSortedKeys": { "type": "boolean" }
+      }
+    },
+
+    // The parameters to a Combine transform and further metadata.
+    "CombinePayload": {
+      "type": "object",
+      "additionalProperties": false,
+
+      "properties": {
+        // The ID of the user-defined CombineFn
+        "combineFn": { "type": "string" }
+      }
+    },
+
+    "WindowIntoPayload": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "windowingStrategy": { "type": "string" }
+      }
+    },
+
+    // A transform node contains pointers to its inputs and outputs, whether
+    // or not it is composite. If it is part of a composite, it contains a pointer
+    // to its parent.
+    "TransformNode": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+
+        // The stable name of the transform application node.
+        "label": { "type": "string" },
```

Comment:
No, this is the stable unique name for identifying the node for e.g. pipeline update.